### DISCUSSION TITLE
How to exclude user repository or organization repository from a search?

### DISCUSSION THREAD

**Discussion by Neustradamus** [community/community#23318] [New to GitHub]
<p>When I do a search on GitHub, I can not exclude a repository (or several) of an user or an organization (or both):</p>
<p>Examples:</p>
<ul>
<li>
<a href="https://github.com/search?q=org%3Agithub+test+-author%3Agithub%2Fgit-msysgit&amp;type=Code" rel="nofollow noopener">https://github.com/search?q=org%3Agithub+test±author%3Agithub%2Fgit-msysgit&amp;type=Code</a><br>
or<br>
- <a href="https://github.com/search?q=org%3Agithub+test+-org%3Agithub%2Fgit-msysgit&amp;type=Code" rel="nofollow noopener">https://github.com/search?q=org%3Agithub+test±org%3Agithub%2Fgit-msysgit&amp;type=Code</a><br>
or<br>
- <a href="https://github.com/search?q=org%3Agithub+test+-user%3Agithub%2Fgit-msysgit&amp;type=Code" rel="nofollow noopener">https://github.com/search?q=org%3Agithub+test±user%3Agithub%2Fgit-msysgit&amp;type=Code</a>
</li>
</ul>
<p>-&gt; github/git-msysgit always in search results.</p>

**Comment by francisfuzz** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/neustradamus">@neustradamus</a> wrote:</p>
</blockquote>
<blockquote>
<p>When I do a search on GitHub, I can not exclude a repository (or several) of an user or an organization (or both):</p>
</blockquote>
<blockquote>
<p>Examples:</p>
<ul>
<li>
<a href="https://github.com/search?q=org%3Agithub+test+-author%3Agithub%2Fgit-msysgit&amp;type=Code" rel="nofollow noopener">https://github.com/search?q=org%3Agithub+test±author%3Agithub%2Fgit-msysgit&amp;type=Code</a><br>
or<br>
- <a href="https://github.com/search?q=org%3Agithub+test+-org%3Agithub%2Fgit-msysgit&amp;type=Code" rel="nofollow noopener">https://github.com/search?q=org%3Agithub+test±org%3Agithub%2Fgit-msysgit&amp;type=Code</a><br>
or<br>
- <a href="https://github.com/search?q=org%3Agithub+test+-user%3Agithub%2Fgit-msysgit&amp;type=Code" rel="nofollow noopener">https://github.com/search?q=org%3Agithub+test±user%3Agithub%2Fgit-msysgit&amp;type=Code</a>
</li>
</ul>
</blockquote>
<blockquote>
<p>-&gt; github/git-msysgit always in search results.</p>
</blockquote>
<p>Rather than negating a specific author, organization, or user, you can exclude a repository by prepending <code>-</code> before <code>repo</code>.</p>
<p>Here’s an example search query:</p>
<ul>
<li><a href="https://github.com/search?q=org%3Agithub+test+-repo%3Agithub%2Fgit-msysgit+is%3Acode&amp;type=Code" title="`org:github is:code test -repo:github/git-msysgit`" rel="nofollow noopener">org:github is:code test -repo:github/git-msysgit</a></li>
</ul>
<p>Here’s our documentation about excluding things from search:</p>
<ul>
<li><a href="https://help.github.com/articles/understanding-the-search-syntax/#exclude-certain-results" rel="nofollow noopener">https://help.github.com/articles/understanding-the-search-syntax/#exclude-certain-results</a></li>
</ul>
<p>I hope this helps!</p>

**Reply by sparr** [reply]
Is there a way to exclude all forks of a repo, or all repos with the same name regardless of which account they are on?

**Reply by ameaninglessname** [reply]
Looks like this syntax doesn't work on `filter` of [notification page](https://github.com/notifications)?

**Reply by PGijsbers** [reply]
I find the same problem;  

* `repo:openml/openml-python` correctly shows things from that repository only
* `-repo:openml/openml-python` does not exclude that repository, instead seems to behave like no filter at all
* `org:openml` correctly shows things from only that organisation
* `-org:openml` does not exclude that organisation, instead seems to behave like no filter at all

<img width="477" alt="image" src="https://github.com/user-attachments/assets/ab624643-2d8d-4b18-aaf0-318367ecbe7b" />
<img width="408" alt="image" src="https://github.com/user-attachments/assets/f202b8ee-c73f-4675-9091-ffe21d73612d" />

**Reply by terrytangyuan** [reply]
+1 this does not work on notifications

**Comment by Neustradamus** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a>: Thanks a lot!</p>
